# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1449516579 18000
#      Mon Dec 07 14:29:39 2015 -0500
# Node ID eed8ec05f1c84231ceafe6d11b3fcdf3a72434ed
# Parent  f428cbe7576f916b0f6310d160d593df292d71fa
always return value from DEFUN functions

* load-path.cc (Fpath): Always return something.
* pager.cc (Fdiary): Likewise.

diff --git a/libinterp/corefcn/load-path.cc b/libinterp/corefcn/load-path.cc
--- a/libinterp/corefcn/load-path.cc
+++ b/libinterp/corefcn/load-path.cc
@@ -2381,16 +2381,18 @@ No checks are made for duplicate element
         "\nOctave's search path contains the following directories:\n\n";
 
       string_vector dirs = load_path::dirs ();
 
       dirs.list_in_columns (octave_stdout);
 
       octave_stdout << "\n";
     }
+
+  return octave_value ();
 }
 
 DEFUN (addpath, args, nargout,
        "-*- texinfo -*-\n\
 @deftypefn  {Built-in Function} {} addpath (@var{dir1}, @dots{})\n\
 @deftypefnx {Built-in Function} {} addpath (@var{dir1}, @dots{}, @var{option})\n\
 Add named directories to the function search path.\n\
 \n\
diff --git a/libinterp/corefcn/pager.cc b/libinterp/corefcn/pager.cc
--- a/libinterp/corefcn/pager.cc
+++ b/libinterp/corefcn/pager.cc
@@ -568,16 +568,18 @@ With no arguments, @code{diary} toggles 
         }
       else
         {
           diary_file = arg;
           write_to_diary_file = true;
           open_diary_file ();
         }
     }
+
+  return octave_value ();
 }
 
 DEFUN (__diaryfile__, , ,
        "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {@var{fname} =} __diaryfile__ ()\n\
 Undocumented internal function\n\
 @end deftypefn")
 {
